### PR TITLE
feat: SDA-1983: implement simplified installer window on SDA

### PR DESCRIPTION
**Pull request by VishwasShashidhar**
## Description
Implement simplified installer window on SDA to allow users to set pod url upon first launch.
See [SDA-1983](https://perzoinc.atlassian.net/browse/SDA-1983) for more details.

## Solution Approach
See [SDA-1983](https://perzoinc.atlassian.net/browse/SDA-1983) for more details.

## Related PRs
#973

### CODE REVIEW

File: src/app/window-handler.ts [hunk 1]
```diff
@@ -160,12 +164,28 @@ export class WindowHandler {
...
         logger.info(`window-handler: initialized spellchecker module with locale ${this.spellchecker.locale}`);
 
         logger.info('window-handler: createApplication mainWinPos: ' + JSON.stringify(this.config.mainWinPos));
 
+        let {isFullScreen, isMaximized} = this.config.mainWinPos ? this.config.mainWinPos : {isFullScreen: false, isMaximized: false};
+
+        this.url = WindowHandler.getValidUrl(this.userConfig.url ? this.userConfig.url : this.globalConfig.url);
+        logger.info(`window-handler: setting url ${this.url} from config file!`);
+
+        if (this.globalConfig.url.startsWith('https://my.symphony.com') && !this.userConfig.url) {
```
Comment (johankwarnmarksymphony) [inline src/app/window-handler.ts:178]: Will this not show the dialog every time for a user that want to use my.symphony.com

Don't you need a flag e.g. showWelcome

Comment (VishwasShashidhar) [reply src/app/window-handler.ts:178]: No @johankwarnmarksymphony. It shows the dialog only if the url is not set in user config file.

And, once a user sets the url, we update that in the user config file, so, this condition won't come into picture after that.

```
ipcMain.on('set-pod-url', async (_event, newPodUrl: string) => {
             await config.updateUserConfig({url: newPodUrl, mainWinPos: { ...this.mainWindow!.getPosition(), ...{ height: 900, width: 900 } } });
             app.relaunch();
             app.exit();
         });
```

File: src/app/config-handler.ts [hunk 2]
```diff
@@ -324,7 +324,9 @@ class Config {
             await app.whenReady();
             await this.readGlobalConfig();
             logger.info(`config-handler: user config doesn't exist! will create new one and update config`);
-            await this.updateUserConfig({ configVersion: app.getVersion().toString(), buildNumber, ...this.globalConfig } as IConfig);
+            const newUserConfig = this.globalConfig as IConfig;
+            delete newUserConfig.url;
```
Comment (KiranNiranjan) [inline src/app/config-handler.ts:328]: This will also remove the URL field from `this.globalConfig`

Comment (VishwasShashidhar) [reply src/app/config-handler.ts:328]: You are right, fixed and pushed a new commit.

File: src/app/config-handler.ts [hunk 3]
```diff
@@ -324,7 +324,7 @@ class Config {
             await app.whenReady();
             await this.readGlobalConfig();
             logger.info(`config-handler: user config doesn't exist! will create new one and update config`);
-            const newUserConfig = this.globalConfig as IConfig;
+            const newUserConfig = JSON.parse(JSON.stringify(this.globalConfig)) as IConfig;
```
Comment (KiranNiranjan) [inline src/app/config-handler.ts:327]:
```suggestion
            const { url, ...rest } = this.globalConfig;
```

**Pull request merged**